src/shuffile.c

```diff
@@ -169,7 +187,16 @@ static void shuffile_read_file(
   if (rank_storage == 0) {
     /* read data from shuffile file */
     kvtree* filedata = kvtree_new();
-    kvtree_read_file(name, filedata);
+    int rc_kvtree = kvtree_read_file(name, filedata);
+    if(rc_kvtree != KVTREE_SUCCESS){
```

Comment:
We need to look at all of these return SHUFFILE_FAILURE inserts in more detail.  For example, this change would lead to a deadlock.  Others might as well, for example, if some ranks pass in a NULL argument but others do not.  We should add an allreduce in places to catch that, or call abort instead of throwing an error and returning.